src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,47 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        validateUserParametersNotNull(user);
+        checkParameters(user);
+        checkExistingUser(user);
```

Comment:
I would create separate method for each verification
```suggestion
        validateLogin(user); // includes check whether such login is used
        validatePassword(user);
        validateAge(user);
```